### ISSUE TITLE
Formatted time cannot be parsed using same format

### ISSUE THREAD

**Issue by mnn** [haskell/time#122]
```hs
formatIso8601WithTime = iso8601DateFormat $ Just "%H:%M%EZ"

formatDateTime :: FormatTime t => t -> Text
formatDateTime = formatTime defaultTimeLocale formatIso8601WithTime >>> T.pack

parseDateTime :: ParseTime t => Text -> t
parseDateTime = T.unpack >>> parseTimeOrError False defaultTimeLocale formatIso8601WithTime
```
Test:
```hs
test_formatDateTime_parseDateTime = do
  cur <- getCurrentTime
  putStrLn $ "cur = " <> show cur
  let formatted = formatDateTime cur
  putStrLn $ T.unpack $ "formatted = " <> formatted
  let parsed = parseDateTime formatted
  putStrLn $ "parsed = " <> show parsed
  assertEqual cur parsed
```
Result:
```
[TEST] UtilsSpec:formatDateTime_parseDateTime (test/UtilsSpec.hs:57)
cur = 2019-10-18 08:59:06.659045819 UTC
formatted = 2019-10-18T08:59Z
parseTimeOrError: no parse of "2019-10-18T08:59Z"
CallStack (from HasCallStack):
  error, called at libraries/time/lib/Data/Time/Format/Parse.hs:117:12 in time-1.8.0.2:Data.Time.Format.Parse
@@@ Error! (0ms)
```

Tested with latest LTS stackage snapshot (lts-14.10, library has version 1.8.0.2). I cannot use nightly snapshot (several libraries are broken).

**Comment by AshleyYakeley** [comment]
It looks like your `%EZ` is the problem. It's not accepted for parsing. Use `%Z` instead, which will parse both `HHMM` and `HH:MM` format.

**Comment by AshleyYakeley** [comment]
I believe this is not a bug.

**Comment by mnn** [comment]
Do I understand correctly, it's intentional that using a specific format X to format a date which then **cannot** be parsed using **same** format X which was used for formatting? 😶

If that's true, it would be good to at least mention such counter-intuitive fact (IMO wart) in docs.

**Comment by AshleyYakeley** [comment]
The idea is that `E` is a modifier for formatting, and has no use for parsing, because parsing accepts both the unmodified and modified format.

Perhaps it would be better to simply ignore the `E` modifier when parsing.

**Comment by wraithm** [comment]
Imho, this is a long-standing wart with the `time` package. I think there was some debate around this in the past iirc. I second the vote to ignore these modifiers when parsing.

**Comment by AshleyYakeley** [comment]
It turns out this was already fixed in time-1.9.3.
